script/DeployV2Base.s.sol

```diff
@@ -28,59 +41,269 @@ abstract contract DeployV2Base is Script, ConfigBase {
         }
     }
 
-    function _deployDeployer() internal {
-        bytes32 salt = __getSalt(SUPER_DEPLOYER_KEY);
-
-        // Predict the address using CREATE2
-        address predictedAddr = vm.computeCreate2Address(salt, keccak256(type(SuperDeployer).creationCode));
-
-        if (predictedAddr.code.length > 0) {
-            console2.log("SuperDeployer already deployed at:", predictedAddr);
-            console2.log("Skipping deployment...");
-        } else {
-            address superDeployer = address(new SuperDeployer{ salt: salt }());
-            console2.log("SuperDeployer deployed at:", superDeployer);
-            require(superDeployer == predictedAddr, "Address mismatch");
-        }
-
-        configuration.deployer = predictedAddr;
-        console2.log("Using SuperDeployer at:", configuration.deployer);
-    }
-
     function _getContract(uint64 chainId, string memory contractName) internal view returns (address) {
         return contractAddresses[chainId][contractName];
     }
 
+    /// @notice Deploy a contract using DeterministicDeployerLib - Nexus style
+    /// @param contractName Name of the contract for logging
+    /// @param chainId Chain ID for tracking
+    /// @param salt Salt for deterministic deployment
+    /// @param creationCode Bytecode with constructor args
+    /// @return deployedAddr Address of the deployed contract
     function __deployContract(
-        ISuperDeployer deployer,
         string memory contractName,
         uint64 chainId,
         bytes32 salt,
         bytes memory creationCode
     )
         internal
-        returns (address)
+        returns (address deployedAddr)
     {
         console2.log("[!] Deploying %s...", contractName);
 
-        // Check if already deployed (Nexus-style pattern)
-        if (deployer.isDeployed(salt)) {
-            address expectedAddr = deployer.getDeployed(salt);
-            console2.log("[!] %s already deployed at:", contractName, expectedAddr);
+        // Predict address first
+        address predictedAddr = DeterministicDeployerLib.computeAddress(creationCode, salt);
+
+        // Check if already deployed using assembly like Nexus
+        uint256 codeSize;
+        assembly {
+            codeSize := extcodesize(predictedAddr)
+        }
+
+        if (codeSize > 0) {
```

Comment:
could simplify this to use `predicatedAddr.code.length > 0` to be consistent with the below require statment.